src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.Visuals.cs

```diff
@@ -0,0 +1,402 @@
+
+				m_hasPreparedContent = true;
+			}
+		}
+
+		private void SizeAndPositionContentInPopup()
+		{
+			var m_tpPopup = _popup;
+
+			double xOffset = 0;
+			double yOffset = 0;
+
+			var windowBounds = Xaml.Window.Current.Bounds;
+
+			var flowDirection = FlowDirection;
+
+			FrameworkElement spBackgroundAsFE = m_tpBackgroundElementPart;
+
+			// Uno note: we're using latest template version (Redstone3). For now we're not supporting legacy templates.
+
+			if (m_placementMode == PlacementMode.EntireControlInPopup)
+			{
+				Height = windowBounds.Height;
+				Width = windowBounds.Width;
+			}
+			else if (m_tpLayoutRootPart != null)
+			{
+				m_tpLayoutRootPart.Height = windowBounds.Height;
+				m_tpLayoutRootPart.Width = windowBounds.Width;
+			}
+
+			// Uno - omitted code related to display regions, which aren't currently supported
+
+			// When the ContentDialog is in the visual tree, the popup offset has added
+			// to it the top-left point of where layout measured and arranged it to.
+			// Since we want ContentDialog to be an overlay, we need to subtract off that
+			// point in order to ensure the ContentDialog is always being displayed in
+			// window coordinates instead of local coordinates.
+			if (m_placementMode == PlacementMode.TransplantedRootInPopup)
+			{
+				GeneralTransform transformToRoot = m_tpPopup.TransformToVisual(null);
+				var offsetFromRoot = (transformToRoot.TransformPoint(default));
+
+				{
+					xOffset =
+						flowDirection == FlowDirection.LeftToRight ?
+						(xOffset - offsetFromRoot.X) :
+						(xOffset - offsetFromRoot.X) * -1;
+				}
+
+				yOffset -= offsetFromRoot.Y;
+			}
+
+			// Set the ContentDialog left and top position.
+			m_tpPopup.HorizontalOffset = xOffset;
+			m_tpPopup.VerticalOffset = yOffset;
+		}
+
+		private void UpdateTitleSpaceVisibility()
+		{
+			if (m_tpTitlePart != null)
+			{
+				var hasTitle = Title != null || TitleTemplate != null;
+				m_tpTitlePart.Visibility = hasTitle ? Visibility.Visible : Visibility.Collapsed;
+			}
+		}
+
+		private void AdjustVisualStateForInputPane()
+		{
+			Rect inputPaneRect = InputPane.GetForCurrentView().OccludedRect;
+
+			if (m_isShowing && inputPaneRect.Height > 0)
+			{
+				// UNO TODO (but in most cases if we're showing a software input pane, the app won't be windowed)
+				//// The rect we get is in screen coordinates, so translate it into client
+				//// coordinates by subtracting our window's origin point (itself translated
+				//// into screen coords) from it.
+				//{
+				//	wf.Point point = { 0, 0 };
+				//	DXamlCore.GetCurrent().ClientToScreen(&point);
+
+				//	inputPaneRect.X -= point.X;
+				//	inputPaneRect.Y -= point.Y;
+				//}
+
+				Rect getElementBounds(FrameworkElement element)
+
+				{
+					GeneralTransform transform = element.TransformToVisual(null);
+					var width = element.ActualWidth;
+					var height = element.ActualHeight;
+
+					return transform.TransformBounds(new Rect(0, 0, width, height));
+				};
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6389897)